File: mobile/frontend/lib/data/enums.dart
// Application enumerations mirroring the backend schema.
// Each enum has a [value] for DB storage and JSON serialization.

enum UserRole {
  admin('admin'),
  supervisor('supervisor'),
  employee('employee');

  final String value;
  const UserRole(this.value);

  static UserRole fromString(String s) => UserRole.values.firstWhere(
    (e) => e.value == s,
    orElse: () => UserRole.employee,
  );

  String get label {
    switch (this) {
      case UserRole.admin:
        return 'Admin';
      case UserRole.supervisor:
        return 'Supervisor';
      case UserRole.employee:
        return 'Employee';
    }
  }
}

enum OperationType {
  receipt('receipt'),
  transfer('transfer'),
  picking('picking'),
  delivery('delivery');

  final String value;
  const OperationType(this.value);

  static OperationType fromString(String s) => OperationType.values.firstWhere(
    (e) => e.value == s,
    orElse: () => OperationType.receipt,
  );

  String get label {
    switch (this) {
      case OperationType.receipt:
        return 'Receipt';
      case OperationType.transfer:
        return 'Transfer';
      case OperationType.picking:
        return 'Picking';
      case OperationType.delivery:
        return 'Delivery';
    }
  }
}

enum OperationStatus {
  pending('pending'),
  inProgress('in_progress'),
  completed('completed'),
  failed('failed');

  final String value;
  const OperationStatus(this.value);

  static OperationStatus fromString(String s) => OperationStatus.values
      .firstWhere((e) => e.value == s, orElse: () => OperationStatus.pending);

  String get label {
    switch (this) {
      case OperationStatus.pending:
        return 'Pending';
      case OperationStatus.inProgress:
        return 'In Progress';
      case OperationStatus.completed:
        return 'Completed';
      case OperationStatus.failed:
        return 'Failed';
    }
  }
}

enum OrderType {
  command('command'),
  preparation('preparation'),
  picking('picking');

  final String value;
  const OrderType(this.value);

  static OrderType fromString(String s) => OrderType.values.firstWhere(
    (e) => e.value == s,
    orElse: () => OrderType.command,
  );

  String get label {
    switch (this) {
      case OrderType.command:
        return 'Command';
      case OrderType.preparation:
        return 'Preparation';
      case OrderType.picking:
        return 'Picking';
    }
  }
}

enum OrderStatus {
  pending('pending'),
  aiGenerated('ai_generated'),
  validated('validated'),
  overridden('overridden'),
  completed('completed');

  final String value;
  const OrderStatus(this.value);

  static OrderStatus fromString(String s) => OrderStatus.values.firstWhere(
    (e) => e.value == s,
    orElse: () => OrderStatus.pending,
  );

  String get label {
    switch (this) {
      case OrderStatus.pending:
        return 'Pending';
      case OrderStatus.aiGenerated:
        return 'AI Generated';
      case OrderStatus.validated:
        return 'Validated';
      case OrderStatus.overridden:
        return 'Overridden';
      case OrderStatus.completed:
        return 'Completed';
    }
  }
}

File: mobile/backend/app/core/enums.py
"""
Application enumerations for the Warehouse Management System.
"""

from enum import Enum


class UserRole(str, Enum):
    """User role within the warehouse system."""
    ADMIN = "admin"
    SUPERVISOR = "supervisor"
    EMPLOYEE = "employee"


class OperationType(str, Enum):
    """Types of warehouse operations."""
    RECEIPT = "receipt"
    TRANSFER = "transfer"
    PICKING = "picking"
    DELIVERY = "delivery"


class OrderType(str, Enum):
    """Types of warehouse orders."""
    COMMAND = "command"
    PREPARATION = "preparation"
    PICKING = "picking"


class OperationStatus(str, Enum):
    """Status of a warehouse operation."""
    PENDING = "pending"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    FAILED = "failed"


class OrderStatus(str, Enum):
    """Status of a warehouse order."""
    PENDING = "pending"
    AI_GENERATED = "ai_generated"
    VALIDATED = "validated"
    OVERRIDDEN = "overridden"
    COMPLETED = "completed"


class DeliveryStatus(str, Enum):
    """Status of a delivery."""
    VALIDATED = "validated"
    FAILED = "failed"
